server/controllers/conversation.js

```diff
@@ -0,0 +1,122 @@
+const Conversation = require("../models/Conversation");
+const Message = require("../models/Message");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /conversation/createConversation
+// @desc create a new conversation
+// @access Public
+exports.createConversation = asyncHandler(async (req, res, next) => {
+    const { content, otherUsers } = req.body;
+
+    if(!content || !otherUsers){
+        res.status(400).send("content or otherUsers can't be null");
+    }
+
+    const newMessage = await Message.create({
+        sender: req.user.id,
+        content,
+    });
+
+    const existConversation = await Conversation.findOne({
+      otherUsers:{ $all: [`${req.user.id}`, `${otherUsers}`]}
+    });
+
+    if(existConversation){
+        existConversation.messages.push(newMessage);
+
+        await existConversation.save();
+
+        res.status(200).json({
+            success: {
+                contactId: otherUsers,
+            }
+        });
+    } else {
+        const conversation = await Conversation.create({
+            otherUsers: [req.user.id, receiver],
+          });
+      
+          conversation.messages.push(message);
+      
+          await conversation.save();
+
+          res.status(200).json({
+              success: {
+                conversation: {
+                  name: existConversation.name,
+                  email: existConversation.email,
+              }
+            },
+          });
+        }
+    });
+
+
+// @route GET /conversation/getAllMessageByConversation
```

Comment:
Isn't this supposed to be `/conversations/getAllMessagesByAConversation`, my pointer is the base route is `/conversations`